plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/PurRepository.java

```diff
@@ -2216,11 +2437,18 @@ public TransMeta loadTransformation( final String transName, final RepositoryDir
         throw new KettleFileException( BaseMessages.getString( PKG,
           "PurRepository.ERROR_0008_TRANSFORMATION_PATH_INVALID", transName ) );
       }
-      RepositoryFile file = pur.getFile( absPath );
-      if ( versionId != null ) {
-        // need to go back to server to get versioned info
-        file = pur.getFileAtVersion( file.getId(), versionId );
+      readWriteLock.readLock().lock();
+      RepositoryFile file;
+      try {
+        file = pur.getFile( absPath );
+        if ( versionId != null ) {
+          // need to go back to server to get versioned info
+          file = pur.getFileAtVersion( file.getId(), versionId );
+        }
+      } finally {
+        readWriteLock.readLock().unlock();
```

Comment:
The Write lock should not be release between this and lower in the same method. The lock should be kept. Make this a write lock.